src/main/java/org/springframework/data/domain/ExampleMatcher.java

```diff
@@ -394,8 +395,8 @@ public boolean isAnyMatching() {
 	@EqualsAndHashCode
 	public static class GenericPropertyMatcher {
 
-		StringMatcher stringMatcher = null;
-		Boolean ignoreCase = null;
+		@Nullable StringMatcher stringMatcher = null;
+		@Nullable Boolean ignoreCase = null;
```

Comment:
Below in ` withMatcher(String propertyPath, GenericPropertyMatcher genericPropertyMatcher)`, the `genericPropertyMatcher.valueTransformer != null` check seems no longer required